src/components/SearchClustersPage.vue

```diff
@@ -91,18 +99,23 @@ export default {
       offset: 0,
       total: 0
     },
-    selectedCluster: undefined
+    paginationList: {
```

Comment:
I would advise to make `paginationList` a computed property for two reasons:
* by design both `totalRows` and `currentPage` must always track changes in page query parameters. Keeping them as a separate object opens a possibility to accidentally changing these values in the code. When they are changed, neither query parameters are updated nor new API request is done. However when other query parameters change, these pagination changes will be used in the API query but again not reflected in the query parameter. I can demonstrate it on a whiteboard tomorrow.
* The `perPage` parameter is duplicated and does not track the `paginationPerPage` property. If it changes higher up in the component tree, the changes are not reflected in the actual pagination.

Reply:
I tried that, but I got stack in an infinite loop probably due to a concurrent computed property (which I removed). 
My idea is to keep the component as simple as possible, to avoid redundancies in computed properties : 
- whenever $route.query change:
   1. update currentPage and other parameters
   2. trigger search according to the new paginationList
I agree, perPage is duplicate. Why was set as a component property? I mean, we won't change it outside of this component.

Reply:
What you describe in `i` and `ii` is already implemented in vue.js as a computed property. Why use a watcher when there is a syntactic sugar readily available that helps reduce boilerplate? Vue.js official documentation also advises to use computed properties over watchers when possible.

Regarding pagination, we won't change it outside now and for this particular component it could be made a constant (in this case it is probably better to declare it as a constant in the component file but outside of the `data` attribute to explicitly indicate that it will never be changed by the component). However if this component is reused and pagination needs to be changed, the way it was done before is the best way to deal with it.

Regarding infinite loop: it is a clear indicator of a problem with one way data flow in the design of the component. I don't know which computed property was leading to an infinite loop but if it was it means that the value of this property was constantly changing which means that either the query parameters were constantly changing or one of the properties was.

Reply:
Fixed. I kept the `paginationPerPage` property for the moment. PaginationList will become a computed property.